<launch>
  <!-- Control Params -->
  <arg name="ctrl_odom_topic" default="/scanmatch_odom"/>
  <arg name="enable_controller" default="true"/>
  <arg name="enable_record" default="false"/>

  <node pkg="ref_gvn_lite" type="ref_gvn_lite_node.py" name="ref_gvn_lite" output="screen">
    <param name="display_nav_path" value="true"/>
    <param name="odom_topic" value="$(arg ctrl_odom_topic)"/>
    <param name="path_topic" value="/astar_path"/>
    <param name="radius_topic" value="/sdf_val"/>
    <param name="gvn_restart_topic" value="/gvn_restart"/>
    <param name="ctrl_freq" value="50.0"/>
    <param name="kg" value="0.5"/>
  </node>

  <group if="$(arg enable_controller)" >
  <!--- Run cone controller as low level velocity controller -->
  <node pkg="erl_clf_cbf_controller" type="clf_cbf_controller_node.py" name="clf_cbf_controller"  output="screen">
    <param name="odom_topic" value="$(arg ctrl_odom_topic)"/>
    <param name="path_topic" value="/astar_path"/>
    <param name="goal_topic" value="/ref_gvn_lite/local_goal"/>
    <param name="ctrl_freq" value="50.0"/>
    <!-- Control limits are set as the same as in control.yaml in jackal_control package -->
    <!-- linear velocity limit m/sec-->
    <param name="v_min" value="-0.15"/>
    <param name="v_max" value="0.15"/>
    <!-- angular velocity limit rad/sec-->
    <param name="w_min" value="-3.14"/>
    <param name="w_max" value="3.14"/>
  </node>

  <node pkg="erl_clf_cbf_controller" type="cvxpy_solver_node.py" name="clf_cbf_cvxpy_solver" output="screen" />
  </group >
</launch>
